src/test.js

```diff
@@ -827,17 +828,14 @@ function internalStart( test ) {
 
 	// Add a slight delay to allow more assertions etc.
 	if ( setTimeout ) {
-		if ( config.timeout ) {
-			clearTimeout( config.timeout );
```

Comment:
clearTimeout tolerates IDs for already-expired timeouts as well as other garbage values. We already don't check it in some places, doing it here as well.